tests/fourier/test_spectrum.py

```diff
@@ -0,0 +1,241 @@
+        assert np.allclose(res_false["x"], range(-2, 3))
+
+    def test_input_gates_not_of_correct_form(self):
+        """Test that an error is thrown if gates marked as encoding gates
+        are not single-parameter gates."""
+
+        dev = qml.device("default.qubit", wires=3)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.1, wires=0, id="x")
+            qml.Rot(0.2, 0.3, 0.4, wires=1, id="x")
+            return qml.expval(qml.PauliZ(wires=0))
+
+        with pytest.raises(ValueError, match="can only consider one-parameter gates"):
+            spectrum(circuit)()
+
+
+def circuit(x, w):
+    """Test circuit"""
+    for l in range(2):
+        for i in range(3):
+            qml.RX(x[i], wires=0, id="x" + str(i))
+            qml.RY(w[l][i], wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 2])
+    qml.RZ(x[0], wires=0, id="x0")
+    return qml.expval(qml.PauliZ(wires=0))
+
+
+expected_result = {
+    "x0": [-3.0, -2.0, -1.0, 0.0, 1.0, 2.0, 3.0],
+    "x1": [-2.0, -1.0, 0.0, 1.0, 2.0],
+    "x2": [-2.0, -1.0, 0.0, 1.0, 2.0],
+}
+
+
+class TestInterfaces:
+    """Test that inputs are correctly identified and spectra computed in
+    all interfaces."""
+
+    def test_integration_autograd(self):
+        """Test that the spectra of a circuit is calculated correctly
+        in the autograd interface."""
+
+        x = pnp.array([1.0, 2.0, 3.0], requires_grad=False)
+        w = pnp.array([[-1, -2, -3], [-4, -5, -6]], requires_grad=True)
+
+        dev = qml.device("default.qubit", wires=3)
+        qnode = qml.QNode(circuit, dev, interface="autograd")
+
+        res = spectrum(qnode)(x, w)
+        for (k1, v1), (k2, v2) in zip(res.items(), expected_result.items()):
+            assert k1 == k2
+            assert v1 == v2
+
+    def test_integration_torch(self):
+        """Test that the spectra of a circuit is calculated correctly
+        in the torch interface."""
+
+        torch = pytest.importorskip("torch")
+        x = torch.tensor([1.0, 2.0, 3.0], requires_grad=True)
+        w = torch.tensor([[-1, -2, -3], [-4, -5, -6]], requires_grad=False)
+
+        dev = qml.device("default.qubit", wires=3)
+        qnode = qml.QNode(circuit, dev, interface="torch")
+
+        res = spectrum(qnode)(x, w)
+        assert res
+        for (k1, v1), (k2, v2) in zip(res.items(), expected_result.items()):
+            assert k1 == k2
+            assert v1 == v2
+
+    def test_integration_tf(self):
+        """Test that the spectra of a circuit is calculated correctly
+        in the tf interface."""
+        tf = pytest.importorskip("tensorflow")
+
+        dev = qml.device("default.qubit", wires=3)
+        qnode = qml.QNode(circuit, dev, interface="tf")
+
+        with tf.GradientTape() as tape:
+            x = tf.Variable([1.0, 2.0, 3.0])
+            w = tf.constant([[-1, -2, -3], [-4, -5, -6]])
+            # the spectrum function has to be called in a tape context
```

Comment:
Maybe worth mentioning this in the `spectrum` documentation, when you discuss how it works on all interfaces

Reply:
Wait, this is now outdated :)